sqlmodel/main.py

```diff
@@ -569,11 +571,26 @@ def model_validate(
         validated = super().model_validate(
             obj, strict=strict, from_attributes=from_attributes, context=context
         )
-        return cls(**{key: value for key, value in validated})
+
+        # remove defaults so they don't get validated
+        data = {}
+        for key, value in validated:
+            field = cls.model_fields[key]
+
+            if (
+                hasattr(field, "default")
+                and field.default is not PydanticUndefined
+                and value == field.default
```

Comment:
We. need to figure out `default_factory` as well I think.

```suggestion
                hasattr(field, "default")
                and field.default is not PydanticUndefined
                and field.default_factory is not None
                and value == field.default
```

Reply:
The default factory is a bit trickier since it is hard to know if the passed value is the output of the default factory or a user provided value. I'm not sure if Pydantic does run field validations in the case it was generated by the default factor, but if it doesn't then we need to avoid SQLModel from running the validation.